src/test/java/com/google/cloud/flink/bigquery/integration/FlinkBQWriteIntegrationTest.java

```diff
@@ -0,0 +1,307 @@
+        srcQueryString
+            + "\n"
+            + "WITH (\n"
+            + "  'connector' = 'bigquery',\n"
+            + "  'format' = 'arrow',\n"
+            + "  'table' = '"
+            + bigqueryWriteTable
+            + "',\n"
+            + "  'selectedFields' = 'word,word_count',\n"
+            + "  'credentialsFile' = '"
+            + System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+            + "' \n"
+            + ")");
+    Table result = tEnv.from(flinkSrcTable);
+    Table datatable = result.select($("word"), $("word_count"));
+    TableResult tableapi = datatable.execute();
+    int count = 0;
+    try (CloseableIterator<Row> it = tableapi.collect()) {
+      while (it.hasNext()) {
+        it.next();
+        count += 1;
+      }
+    }
+    datatable.execute();
+    return count;
+  }
+
+  public void testWriteToBigQuery(String bigqueryReadTable, String bigqueryWriteTable)
+      throws Exception {
+    final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    env.setParallelism(1); // source only supports parallelism of 1
+    final StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+    String flinkSrcTable = "FlinkSrcTable";
+    String flinkSinkTable = "FlinkSinkTable";
+    String srcQueryString = "CREATE TABLE " + flinkSrcTable + " (word STRING , word_count BIGINT)";
+    tEnv.executeSql(
+        srcQueryString
+            + "\n"
+            + "WITH (\n"
+            + "  'connector' = 'bigquery',\n"
+            + "  'format' = 'arrow',\n"
+            + "  'table' = '"
+            + bigqueryReadTable
+            + "',\n"
+            + "  'filter' = 'word_count > 500',\n"
+            + "  'credentialsFile' = '"
+            + System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+            + "' ,\n"
+            + "  'selectedFields' = 'word,word_count' \n"
+            + ")");
+    Table sourceTable = tEnv.from(flinkSrcTable);
+    Table sourceResultTable =
+        sourceTable.where($("word_count").isGreaterOrEqual(500)).select($("word"), $("word_count"));
+    BigQueryTableSink tableSink = new BigQueryTableSink(sourceResultTable, bigqueryWriteTable);
+
+    tEnv.registerTableSink(flinkSinkTable, tableSink);
+
+    TableResult sinkResult = sourceResultTable.executeInsert(flinkSinkTable, true);
+    JobExecutionResult jobExecutionResult =
+        sinkResult
+            .getJobClient()
+            .get()
+            .getJobExecutionResult(Thread.currentThread().getContextClassLoader())
+            .get();
+  }
+
+  @Test
+  public void testPartitionHourly() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.stackoverflow.posts_questions";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "hourlyPartitioned";
+    String filter = "view_count=300 and owner_user_id<5000";
+    testPartition("HOUR", bigqueryReadTable, bigqueryWriteTable, filter);
+    Thread.sleep(120 * 1000);
```

Comment:
why the sleep is needed?

Reply:
Takes some amount of time to write the data. Same wait time found in spark connector also